Winleafs-Server/Winleafs.Server.Services/SpotifyService.cs

```diff
@@ -0,0 +1,181 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Newtonsoft.Json;
+using SpotifyAPI.Web;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+using Winleafs.Server.Models.Models;
+using Winleafs.Server.Services.DTO;
+using Winleafs.Server.Services.Exceptions;
+using Winleafs.Server.Services.Helpers;
+using Winleafs.Server.Services.Interfaces;
+
+namespace Winleafs.Server.Services
+{
+    public class SpotifyService : ISpotifyService
```

Comment:
Something to keep in mind is the rate limit for the Spotify API.
I doubt we will hit it but its good to implement ahead of time.

Reply:
For now, I will add specific logging for status code 429, but I will also introduce exceptions for failed requests